### PR TITLE
GH actions for new tests

### PR DESCRIPTION
**Pull request by uncleDecart**
This PR creates GH actions to run separated tests. It's still WIP, because I want to test this workflows with runners provided by LF-EDGE

### CODE REVIEW

**Comment by uncleDecart** [comment]
cc: @yash-zededa

**Comment by uncleDecart** [comment]
@milan-zededa @eriknordmark @yash-zededa

#### TL;DR
I think we can merge this workflow and create another PR to swap eden integration tests with it once we are ready together with EVE repo once we decide what we will do with runners and fix as many tests as possible

#### Time improvement 
Our baseline (github-hosted runners or LF-EDGE sponsored runners) are around 4-5 hours to complete eden integration test flow.
I ran current workflow (eden integration test) on dedicated runner and time taking to run whole workflow was 2 hours 10 minutes [link](https://github.com/uncleDecart/eden/actions/runs/5003281246?pr=1).
I also did some tests on [local runner](https://github.com/uncleDecart/eden/actions/runs/5197858608) as well as on [buildjet](https://github.com/uncleDecart/eden/actions/runs/5198733211?pr=4).
Proposed workflow takes around 1.5 hours to complete with 35 minutes being smoke tests (4 of them running in parallel) which is better than 2 hours.

#### Errors during test execution
ZFS+TPM on self-hosted machine always fails for some reason, I was able to run TPM+ZFS test suite successfully on Github-hosted runner though [link](https://github.com/uncleDecart/eden/actions/runs/5188159951). Sometimes for some reason SSH keys are not appearing on EVE side, so it's floating bug.
I think it makes sense to investigate test failures in new workflow first trying to debug them locally.

File: .github/workflows/publish_logs.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
+name: Collect, count and publish logs
```
Comment (milan-zededa) [inline .github/workflows/publish_logs.yml:1]: Isn't this a duplicate of `.github/actions/publish-logs/action.yml`?

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    needs: smoke
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Get code
+        uses: actions/checkout@v3
+      - name: Run Networking tests
+        uses: ./.github/actions/run-eden-test
+        with:
+          file_system: "ext4"
+          tpm_enabled: false
```
Comment (milan-zededa) [inline .github/workflows/test.yml:36]: I wonder if we should make `tpm_enabled` as enabled for suites that do not use matrix here. I believe that most customer devices do have TPM module.

Comment (milan-zededa) [inline .github/workflows/test.yml:36]: I wonder if we should make `tpm_enabled` as `true` for suites that do not use matrix here. I believe that most customer devices do have TPM module.

Comment (uncleDecart) [reply .github/workflows/test.yml:36]: I thought that networking test suite does not depend on tpm or file system at all, or am I wrong?

File: .github/workflows/publish_logs.yml (hunk 1 above)
Comment (milan-zededa) [reply .github/workflows/publish_logs.yml:1]: I mean, could you just do `uses: ./.github/actions/publish-logs` here to avoid code duplication?

Comment (uncleDecart) [reply .github/workflows/publish_logs.yml:1]: My bad, I duplicated code, will remove it :)

File: .github/workflows/test.yml (hunk 2 above)
Comment (uncleDecart) [reply .github/workflows/test.yml:36]: Fair, I'll change it

File: .github/workflows/publish_logs.yml (hunk 1 above)
Comment (uncleDecart) [reply .github/workflows/publish_logs.yml]: @milan-zededa done

File: .github/workflows/test.yml (hunk 2 above)
Comment (uncleDecart) [reply .github/workflows/test.yml]: @milan-zededa done

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+      - name: Run Smoke tests
+        uses: ./.github/actions/run-eden-test
+        with:
+          file_system: ${{ matrix.file_system }}
+          tpm_enabled: ${{ matrix.tpm }}
+          suite: "smoke.tests.txt"
+
+  networking: 
+    name: Networking test suite
+    needs: smoke
```
Comment (milan-zededa) [inline .github/workflows/test.yml:27]: > needs: smoke

Seems our new workflows have nicotine addiction :D

**Review by milan-zededa — APPROVED** [review]

File: .github/actions/publish-logs/action.yml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        echo "::group::Segfaults"
+        echo "$(jq -s 'map(select(.content | contains("segfault at")))' trace.log)"|tee segfaults.log
+        [ "$(jq length segfaults.log)" -gt 0 ] && echo "::warning::segfaults found, you can see them in Log counting->Segfaults section"
+        echo "::endgroup::"
+      shell: bash
+    - name: Store raw test results
+      if: ${{ always() }}
+      uses: actions/upload-artifact@v3
+      with:
+        name: eden-report-tpm-${{ matrix.tpm }}-${{ matrix.fs }}
```
Comment (giggsoff) [inline .github/actions/publish-logs/action.yml:41]: Will we have `matrix` defined for e.g. `networking` test? Seems we should have additional `inputs`, probably, just a name.

File: .github/actions/run-eden-test/action.yml [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    - name: Setup Environment
+      uses: ./.github/actions/setup-environment
+      with:
+        file_system: ${{ inputs.file_system }}
+        tpm_enabled: ${{ inputs.tpm_enabled }}
+    - name: Run tests
+      run: EDEN_TEST_STOP=n ./eden test ./tests/workflow -s ${{ inputs.suite }} -v debug
+      shell: bash
+    - name: Collect logs
+      uses: ./.github/actions/publish-logs
```
Comment (giggsoff) [inline .github/actions/run-eden-test/action.yml:26]: We should use `if: always()` here.
Actually I cannot find any logs in your [sample](https://github.com/uncleDecart/eden/actions/runs/5197858608), am I missing something?

File: .github/actions/publish-logs/action.yml (hunk 4 above)
Comment (uncleDecart) [reply .github/actions/publish-logs/action.yml:41]: Yes, missed that. Thank you!

Comment (giggsoff) [reply .github/actions/publish-logs/action.yml]: Seems we should call the action using `with` option after inputs added.

File: .github/actions/run-eden-test/action.yml (hunk 5 above)
Comment (uncleDecart) [reply .github/actions/run-eden-test/action.yml]: done

File: .github/actions/publish-logs/action.yml (hunk 4 above)
Comment (uncleDecart) [reply .github/actions/publish-logs/action.yml]: done

**Review by giggsoff — APPROVED** [review]
